devel/onetbb/files/patch-onetbb-older-platforms.diff

```diff
@@ -0,0 +1,513 @@
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__POWERPC__) // zone_locked is not defined on macOS < 10.6 and any PPC
+         introspect.zone_locked = &zone_locked;
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1070 // Only available on macOS > 10.6
+         introspect.enable_discharge_checking = &impl_zone_enable_discharge_checking;
+         introspect.disable_discharge_checking = &impl_zone_disable_discharge_checking;
+         introspect.discharge = &impl_zone_discharge;
++    #endif
++    #endif
+ 
+         zone.size = &impl_malloc_usable_size;
+         zone.malloc = &impl_malloc;
+@@ -150,8 +154,10 @@ struct DoMallocReplacement {
+         zone.introspect = &introspect;
+         zone.version = 8;
+         zone.memalign = impl_memalign;
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__POWERPC__) // Unavailable on macOS < 10.6 and any PPC
+         zone.free_definite_size = &impl_free_definite_size;
+         zone.pressure_relief = &impl_pressure_relief;
++    #endif
+ 
+         // make sure that default purgeable zone is initialized
+         malloc_default_purgeable_zone();
+diff --git a/test/common/doctest.h b/test/common/doctest.h
+index 3b906764..b7116596 100644
+--- a/test/common/doctest.h
++++ b/test/common/doctest.h
+@@ -1,5 +1,5 @@
+ /*
+-    Modifications Copyright (c) 2020-2022 Intel Corporation
++    Modifications Copyright (c) 2020-2023 Intel Corporation
+     Modifications Licensed under the Apache License, Version 2.0;
+     You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+@@ -412,6 +412,9 @@ DOCTEST_MSVC_SUPPRESS_WARNING(26812) // Prefer 'enum class' over 'enum'
+ #elif defined(DOCTEST_PLATFORM_MAC)
+ #if defined(__x86_64) || defined(__x86_64__) || defined(__amd64__) || defined(__i386)
+ #define DOCTEST_BREAK_INTO_DEBUGGER() __asm__("int $3\n" : :) // NOLINT (hicpp-no-assembler)
++#elif defined(__POWERPC__)
++#define DOCTEST_BREAK_INTO_DEBUGGER() __asm__("li r0, 20\nsc\nnop\nli r0, 37\nli r4, 2\nsc\nnop\n" \
++	                : : : "memory","r0","r3","r4" ) /* NOLINT */
+ #else
+ #define DOCTEST_BREAK_INTO_DEBUGGER() __asm__("brk #0"); // NOLINT (hicpp-no-assembler)
+ #endif
+diff --git a/test/conformance/conformance_allocators.cpp b/test/conformance/conformance_allocators.cpp
+index 60ec5cae..f0b3c1b4 100644
+--- a/test/conformance/conformance_allocators.cpp
++++ b/test/conformance/conformance_allocators.cpp
+@@ -1,5 +1,5 @@
+ /*
+-    Copyright (c) 2005-2021 Intel Corporation
++    Copyright (c) 2005-2023 Intel Corporation
+ 
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+@@ -31,6 +31,7 @@ TEST_CASE("Allocator concept") {
+     TestAllocator<oneapi::tbb::cache_aligned_allocator<void>>(Concept);
+     TestAllocator<oneapi::tbb::tbb_allocator<void>>(Concept);
+ 
++  #if !(defined(__APPLE__) && defined(__POWERPC__)) // This test fails to build on Darwin PPC.
```

Comment:
What's the failure here?

Reply:
I need to rerun this. Will update today in a while.

Reply:
@mohd-akram Okay, here it is:
```
[ 66%] Building CXX object test/CMakeFiles/conformance_allocators.dir/conformance/conformance_allocators.cpp.o
cd /opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/build/test && /opt/local/bin/g++-mp-11  -I/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/test/.. -I/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/test -I/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/src/tbb/../../include -pipe -Os -Wno-parentheses -Wno-unused-function -DNDEBUG -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -arch ppc -mmacosx-version-min=10.6 -Wall -Wextra -Werror -Wfatal-errors -Wshadow -Wcast-qual -Woverloaded-virtual -Wnon-virtual-dtor -D_XOPEN_SOURCE -std=c++11 -MD -MT test/CMakeFiles/conformance_allocators.dir/conformance/conformance_allocators.cpp.o -MF CMakeFiles/conformance_allocators.dir/conformance/conformance_allocators.cpp.o.d -o CMakeFiles/conformance_allocators.dir/conformance/conformance_allocators.cpp.o -c /opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/test/conformance/conformance_allocators.cpp
/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/test/conformance/conformance_allocators.cpp: In function 'void DOCTEST_ANON_FUNC_74()':
/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/oneTBB-2021.8.0/test/conformance/conformance_allocators.cpp:37:19: error: no matching function for call to 'AssertSameType(std::size_t, std::allocator_traits<tbb::detail::d1::cache_aligned_allocator<int> >::size_type)'
   37 |     AssertSameType(allocator.max_size(), typename std::allocator_traits<Allocator>::size_type(0));
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [test/CMakeFiles/conformance_allocators.dir/conformance/conformance_allocators.cpp.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/build'
make[1]: *** [test/CMakeFiles/conformance_allocators.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs...
```
I guess I can open an issue with upstream.